src/utils/positionUtils.ts

```diff
@@ -389,34 +426,41 @@ export const calcTradeFee = ({
   const takerProportionalFee = Big6Math.div(Big6Math.mul(notional, takerProportionalFeeNumerator), takerFee.scale)
 
   const takerLinearFee = Big6Math.mul(notional, takerFee.linearFee)
-  const subtractiveFee = Big6Math.mul(takerLinearFee, referralFee)
+  const notionalTakerFee = Big6Math.mul(marketTakerFee, notional)
 
-  const marketFee = Big6Math.mul(takerLinearFee - subtractiveFee, marketTakerFee)
-  const tradeFee = subtractiveFee + marketFee
+  const tradeFee = notionalTakerFee + takerProportionalFee
+  const referralFeeNotional = Big6Math.mul(tradeFee, referralFee)
   const feeBasisPoints = !Big6Math.isZero(tradeFee) ? Big6Math.div(tradeFee, notional) : 0n
-  const tradeImpact = takerLinearFee + takerProportionalFee + takerAdiabaticFee - tradeFee
+  // TODO: Check - Old impact calc subtracts trade fee
```

Comment:
Yep the new calculation is correct since now they're two totally separate parameters